R/auth.R

```diff
@@ -10,43 +10,27 @@
 #' please consider [registering for an API
 #' key](https://api.delphi.cmu.edu/epidata/admin/registration_form).
 #'
-#' API keys are strings and can be set in two ways. If the environment variable
-#' `DELPHI_EPIDATA_KEY` is set, it will be used automatically. Environment
-#' variables can be set either in the shell or by editing your `.Renviron` file,
-#' which will ensure the setting applies to all R sessions. See `?Startup` for a
-#' description of `Renviron` files and where they can be placed.
-#'
-#' Alternately, the API key can be set from within a session by using
-#' `set_api_key()`, which sets the R option `delphi.epidata.key` using
-#' `options()`. If this option is set, it is used in preference to the
-#' environment variable, so you may change keys within an R session. R options
-#' are not preserved between sessions, so `set_api_key()` must be run every time
-#' you open R. Alternately, you can have R set the option at startup by adding
-#' it to your `.Rprofile`; see `?Startup` for a description of `Rprofile` files
-#' and where they can be placed.
+#' API keys are strings read from the environment variable `DELPHI_EPIDATA_KEY`.
+#' We recommend setting your key with `set_api_key()`, which will modify your
+#' `.Renviron` file and will persist your key across R sessions (see `?Startup`
+#' for a description of `.Renviron` files). Alternatively, you can modify the
+#' environment variable in the shell or with `Sys.setenv()`, but these will not
+#' persist across sessions.
```

Comment:
From a skim, it looks like after `set_api_key()` will not set the API key for the current session, only save to the `.Renviron`.  Can this be named `save_api_key()` or maybe also set the key for the current session?

Reply:
(Or clarify this in the docs)

Reply:
Good point, I can rename

Reply:
Some points about setting key automatically:

- don't know what the person wrote in .Renviron
- can't trigger an .Renviron reload with restarting R
- if the key is passed as an argument, is it safe to manually edit `.Renviron` for the user with `write(paste("DELPHI_EPIDATA_KEY=", key), file=environ_path, append=TRUE)` ?
- if the key is passed as an argument, the key will now be in the user's Rhistory